packages/react-components/src/components/Notifications.tsx

```diff
@@ -267,10 +266,29 @@ export const Notifications = (props: NotificationsProps): JSX.Element => {
   const localeStrings = useLocale().strings.notifications;
   const strings = props.strings ?? /* @conditional-compile-remove(notifications) */ localeStrings;
   const maxNotificationsToShow = props.maxNotificationsToShow ?? 2;
-  const [activeNotifications, setActiveNotifications] = useState<ActiveNotification[]>(props.activeNotifications);
+
+  const trackDismissedNotificationsInternally = !props.onDismissNotification;
+
+  // Timestamp for when this comopnent is first mounted.
+  // Never updated through the lifecycle of this component.
+  const mountTimestamp = useRef(new Date(Date.now()));
+
+  const [dismissedNotifications, setDismissedNotifications] = useState<DismissedNotification[]>([]);
+
+  // dropDismissalsForInactiveNotifications only returns a new object if `dismissedErrors` actually changes.
+  // Without this behaviour, this `useEffect` block would cause a render loop.
   useEffect(() => {
-    setActiveNotifications(props.activeNotifications);
-  }, [props.activeNotifications]);
+    trackDismissedNotificationsInternally &&
+      setDismissedNotifications(
+        dropDismissalsForInactiveNotifications(props.activeNotifications, dismissedNotifications)
```

Comment:
Can you provide some details why we have a dismissedAt and a dismissedNotifications Array?
Is it not easier to immediately pop the notification from the list of active notifications? Do we need to keep account of which notifications are dismissed? 

Let me know if I'm missing something here, I feel I'm missing a piece of the puzzle.

Reply:
It's the same logic with error bar. I mainly just copied it over. The idea is to track the dismissed errors so it won't pop up constantly